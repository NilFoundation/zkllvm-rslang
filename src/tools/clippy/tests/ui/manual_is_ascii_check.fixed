// run-rustfix

#![allow(unused, dead_code)]
#![warn(clippy::manual_is_ascii_check)]

fn main() {
    assert!('x'.is_ascii_lowercase());
    assert!('X'.is_ascii_uppercase());
    assert!(b'x'.is_ascii_lowercase());
    assert!(b'X'.is_ascii_uppercase());

    let num = '2';
    assert!(num.is_ascii_digit());
    assert!(b'1'.is_ascii_digit());
    assert!('x'.is_ascii_alphabetic());

    assert!(matches!('x', 'A'..='Z' | 'a'..='z' | '_'));
}

#[clippy::msrv = "1.23"]
fn msrv_1_23() {
    assert!(matches!(b'1', b'0'..=b'9'));
    assert!(matches!('X', 'A'..='Z'));
    assert!(matches!('x', 'A'..='Z' | 'a'..='z'));
}

#[clippy::msrv = "1.24"]
fn msrv_1_24() {
    assert!(b'1'.is_ascii_digit());
    assert!('X'.is_ascii_uppercase());
    assert!('x'.is_ascii_alphabetic());
}

#[clippy::msrv = "1.46"]
fn msrv_1_46() {
    const FOO: bool = matches!('x', '0'..='9');
}

#[clippy::msrv = "1.47"]
fn msrv_1_47() {
    const FOO: bool = 'x'.is_ascii_digit();
}
