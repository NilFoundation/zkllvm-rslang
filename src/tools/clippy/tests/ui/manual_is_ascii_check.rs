// run-rustfix

#![allow(unused, dead_code)]
#![warn(clippy::manual_is_ascii_check)]

fn main() {
    assert!(matches!('x', 'a'..='z'));
    assert!(matches!('X', 'A'..='Z'));
    assert!(matches!(b'x', b'a'..=b'z'));
    assert!(matches!(b'X', b'A'..=b'Z'));

    let num = '2';
    assert!(matches!(num, '0'..='9'));
    assert!(matches!(b'1', b'0'..=b'9'));
    assert!(matches!('x', 'A'..='Z' | 'a'..='z'));

    assert!(matches!('x', 'A'..='Z' | 'a'..='z' | '_'));
}

#[clippy::msrv = "1.23"]
fn msrv_1_23() {
    assert!(matches!(b'1', b'0'..=b'9'));
    assert!(matches!('X', 'A'..='Z'));
    assert!(matches!('x', 'A'..='Z' | 'a'..='z'));
}

#[clippy::msrv = "1.24"]
fn msrv_1_24() {
    assert!(matches!(b'1', b'0'..=b'9'));
    assert!(matches!('X', 'A'..='Z'));
    assert!(matches!('x', 'A'..='Z' | 'a'..='z'));
}

#[clippy::msrv = "1.46"]
fn msrv_1_46() {
    const FOO: bool = matches!('x', '0'..='9');
}

#[clippy::msrv = "1.47"]
fn msrv_1_47() {
    const FOO: bool = matches!('x', '0'..='9');
}
