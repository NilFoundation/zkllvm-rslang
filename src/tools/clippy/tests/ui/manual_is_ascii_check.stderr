error: manual check for common ascii range
  --> $DIR/manual_is_ascii_check.rs:7:13
   |
LL |     assert!(matches!('x', 'a'..='z'));
   |             ^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `'x'.is_ascii_lowercase()`
   |
   = note: `-D clippy::manual-is-ascii-check` implied by `-D warnings`

error: manual check for common ascii range
  --> $DIR/manual_is_ascii_check.rs:8:13
   |
LL |     assert!(matches!('X', 'A'..='Z'));
   |             ^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `'X'.is_ascii_uppercase()`

error: manual check for common ascii range
  --> $DIR/manual_is_ascii_check.rs:9:13
   |
LL |     assert!(matches!(b'x', b'a'..=b'z'));
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `b'x'.is_ascii_lowercase()`

error: manual check for common ascii range
  --> $DIR/manual_is_ascii_check.rs:10:13
   |
LL |     assert!(matches!(b'X', b'A'..=b'Z'));
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `b'X'.is_ascii_uppercase()`

error: manual check for common ascii range
  --> $DIR/manual_is_ascii_check.rs:13:13
   |
LL |     assert!(matches!(num, '0'..='9'));
   |             ^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `num.is_ascii_digit()`

error: manual check for common ascii range
  --> $DIR/manual_is_ascii_check.rs:14:13
   |
LL |     assert!(matches!(b'1', b'0'..=b'9'));
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `b'1'.is_ascii_digit()`

error: manual check for common ascii range
  --> $DIR/manual_is_ascii_check.rs:15:13
   |
LL |     assert!(matches!('x', 'A'..='Z' | 'a'..='z'));
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `'x'.is_ascii_alphabetic()`

error: manual check for common ascii range
  --> $DIR/manual_is_ascii_check.rs:29:13
   |
LL |     assert!(matches!(b'1', b'0'..=b'9'));
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `b'1'.is_ascii_digit()`

error: manual check for common ascii range
  --> $DIR/manual_is_ascii_check.rs:30:13
   |
LL |     assert!(matches!('X', 'A'..='Z'));
   |             ^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `'X'.is_ascii_uppercase()`

error: manual check for common ascii range
  --> $DIR/manual_is_ascii_check.rs:31:13
   |
LL |     assert!(matches!('x', 'A'..='Z' | 'a'..='z'));
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `'x'.is_ascii_alphabetic()`

error: manual check for common ascii range
  --> $DIR/manual_is_ascii_check.rs:41:23
   |
LL |     const FOO: bool = matches!('x', '0'..='9');
   |                       ^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `'x'.is_ascii_digit()`

error: aborting due to 11 previous errors

